xls/passes/select_simplification_pass.cc

```diff
@@ -964,6 +965,40 @@ absl::StatusOr<bool> SimplifyNode(Node* node, const QueryEngine& query_engine,
     }
   }
 
+  // Specialize the arms of the select based on the selector value which chooses
+  // the arm.
+  if (node->Is<Select>()) {
+    XLS_ASSIGN_OR_RETURN(
+        std::unique_ptr<PostDominatorAnalysis> post_dominator_analysis,
+        PostDominatorAnalysis::Run(node->function()));
+    Select* sel = node->As<Select>();
+    int64 selr_width = sel->selector()->BitCountOrDie();
+    bool changed = false;
+    for (int64 selr_value = 0; selr_value < sel->cases().size(); ++selr_value) {
+      Node* arm = sel->cases()[selr_value];
+      // Skip any cases where the select arm is used other than just in the
+      // select statement
+      if (arm->users().size() != 1) {
+        continue;
+      }
+      std::vector<Node*> users(sel->selector()->users().begin(),
+                               sel->selector()->users().end());
+      for (Node* selr_user : users) {
```

Comment:
Using `sel->selector()->users()` directly in the range based for loop here is what was causing the container overflow.

I'm not exactly sure why that is, but I noticed that a call to `users()` elsewhere in the code was being collected into a vector first before being iterated through and that seems to fix it.